kubernetes/resource_kubernetes_cluster_role_test.go

```diff
@@ -182,6 +182,52 @@ func TestAccKubernetesClusterRole_aggregationRuleBasic(t *testing.T) {
 	})
 }
 
+func TestAccKubernetesClusterRole_aggregationRuleMultiple(t *testing.T) {
+	var conf api.ClusterRole
+	resourceName := "kubernetes_cluster_role.test"
+	name := acctest.RandomWithPrefix("tf-acc-test")
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		IDRefreshName:     resourceName,
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckKubernetesClusterRoleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKubernetesClusterRoleConfig_aggRule(name),
```

Comment:
Thank you for adding the acceptance test, however, this one is failing:
```
=== RUN   TestAccKubernetesClusterRole_aggregationRuleMultiple
    resource_kubernetes_cluster_role_test.go:189: Step 1/2 error: Check failed: 2 errors occurred:
                * Check 3/5 error: kubernetes_cluster_role.test: Attribute 'aggregation_rule.0.cluster_role_selectors.#' expected "2", got "1"
                * Check 5/5 error: kubernetes_cluster_role.test: Attribute 'aggregation_rule.0.cluster_role_selectors.1.match_labels.bar' not found
```

As a configuration here you use `testAccKubernetesClusterRoleConfig_aggRule` and it has only one `cluster_role_selectors`. Since this config is used in some other tests, let's create a new one `testAccKubernetesClusterRoleConfig_aggRuleMultiple` with a simplified config in there and use it instead. Something like this:

```
func testAccKubernetesClusterRoleConfig_aggRuleMultiple(name string) string {
	return fmt.Sprintf(`resource "kubernetes_cluster_role" "test" {
  metadata {
    name = "%s"
  }

  aggregation_rule {
    cluster_role_selectors {
      match_labels = {
        foo = "bar"
      }
    }
    cluster_role_selectors {
      match_labels = {
        bar = "foo"
      }
    }  
  }
}
`, name)
}
```

In that case, we can also take out the third step from the test and has fewer checks in the first one. Something like this:
```
resource.TestCheckResourceAttr(resourceName, "aggregation_rule.#", "1"),
resource.TestCheckResourceAttr(resourceName, "aggregation_rule.0.cluster_role_selectors.#", "2"),
resource.TestCheckResourceAttr(resourceName, "aggregation_rule.0.cluster_role_selectors.0.match_labels.foo", "bar"),
resource.TestCheckResourceAttr(resourceName, "aggregation_rule.0.cluster_role_selectors.1.match_labels.bar", "foo"),
```

And thank you for taking care of this issue!

Reply:
Thanks for the feedback @arybolovlev, I'm not sure why it ever passed locally; I've updated it now with two new configurations, one for create and another for modify.